src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.IO.Abstractions;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.Unittests
+{
+    [TestClass]
+    public class ConfigFileWatcherUnitTests
+    {
+        /// <summary>
+        /// Use the file system to add a file that matches the format of our config file.
+        /// Use that to create the needed config loader and provider, and then get the
+        /// actual config object from the provider, instantiating our config file watcher
+        /// in the process. Modify that file we created originally and verify we reload
+        /// the config correctly for the desired RuntimeOptions.
```

Comment:
nit: The summary reads more like all the steps performed in the steps. Not really needed here - can include within the test as comments. Summary should be around what we are testing here - the hot reload successfully updates the runtime section.